src/plugins/vis_type_xy/public/utils/render_all_series.tsx

```diff
@@ -0,0 +1,197 @@
+import { SeriesParam, VisConfig, FakeParams, Aspect } from '../types';
+
+/**
+ * Matches vislib curve to elastic charts
+ * @param type curve type
+ */
+const getCurveType = (type?: 'linear' | 'cardinal' | 'step-after'): CurveType => {
+  switch (type) {
+    case 'cardinal':
+      return CurveType.CURVE_NATURAL;
+    case 'step-after':
+      return CurveType.CURVE_STEP_AFTER;
+    case 'linear':
+    default:
+      return CurveType.LINEAR;
+  }
+};
+
+const getXAccessor = (xAspect: Aspect): Accessor | AccessorFn => {
+  if (xAspect.accessor) {
+    if (xAspect.aggType === 'date_range' && xAspect.formatter) {
+      const formatter = xAspect.formatter;
+      const accessor = xAspect.accessor;
+      return (d) => {
+        const v = d[accessor];
+        if (!v) {
+          return;
+        }
+        const f = formatter(v);
+        return f;
+      };
+    }
+
+    return xAspect.accessor;
+  }
+
+  return () => (xAspect.params as FakeParams)?.defaultValue;
+};
+
+/**
+ * Renders chart Line, Area or Bar series
+ * @param config
+ * @param seriesParams
+ * @param data
+ * @param getSeriesName
+ * @param getSeriesColor
+ */
+export const renderAllSeries = (
+  { aspects, yAxes, xAxis, showValueLabel, enableHistogramMode, fittingFunction }: VisConfig,
+  seriesParams: SeriesParam[],
+  data: DatatableRow[],
+  getSeriesName: (series: XYChartSeriesIdentifier) => SeriesName,
+  getSeriesColor: SeriesColorAccessorFn,
+  timeZone: string
+) => {
+  const xAccessor = getXAccessor(aspects.x);
+
+  return seriesParams.map(
+    ({
+      show,
+      valueAxis: groupId,
+      data: { id: paramId },
+      lineWidth: strokeWidth,
+      showCircles,
+      drawLinesBetweenPoints,
+      mode,
+      interpolate,
+      type,
+    }) => {
+      const yAspect = aspects.y.find(({ aggId }) => aggId === paramId);
+
+      if (!show || !yAspect || yAspect.accessor === null) {
+        return null;
+      }
+
+      const id = `${type}-${yAspect.accessor}`;
+
+      const splitSeriesAccessors = aspects.series
+        ? (compact(aspects.series.map(({ accessor }) => accessor)) as string[])
+        : [];
+      const yAxisScale = yAxes.find(({ groupId: axisGroupId }) => axisGroupId === groupId)?.scale;
+      const stackAccessors =
+        mode === 'stacked' || yAxisScale?.mode === 'percentage' ? ['__any_value__'] : undefined;
+      const stackMode = yAxisScale?.mode === 'normal' ? undefined : yAxisScale?.mode;
+
+      switch (type) {
+        case ChartType.Histogram:
+          return (
+            <BarSeries
+              key={id}
+              id={id}
+              name={getSeriesName}
+              color={getSeriesColor}
+              tickFormat={yAspect.formatter}
+              groupId={groupId}
+              xScaleType={xAxis.scale.type}
+              yScaleType={yAxisScale?.type}
+              xAccessor={xAccessor}
+              yAccessors={[yAspect.accessor]}
+              markSizeAccessor={aspects.z?.accessor ?? undefined}
```

Comment:
You are correct, this has no impact on the bar charts. That said, vislib _does_ still show this value in the tooltip as if it were visible but still doesn't change the chart appearance.

This might be something we should refine later to disable certain metrics when a given type of chart is being used.

Reply:
11cacefe2f1a71ce14122992cb301abd600767b4